koku/api/query_handler.py

```diff
@@ -321,7 +321,7 @@ def filter_to_order_by(self, parameters):  # noqa: C901
 
         """
         for group_by_key in parameters.parameters.get("group_by", {}):
-            for group_by_value in parameters.parameters["group_by"][group_by_key]:
+            for group_by_value in parameters.parameters.get("group_by", {})[group_by_key]:
```

Comment:
I have a question. If the group_by defaults to the `{}`, directly accessing the `group_by_key` will cause a key error - Is this okay or should we do another get defaulted to another `{}`?

Reply:
The default return value for `.get()` is `None`. 

The full signature of the `dict.get()` method is `.get(value, default=None)` and whatever the default is gets returned if an `AttributeError` is thrown by trying to access a key that's not in the dict.

Reply:
@abaiken correct me if I am wrong, but I think your point was that if he does a `.get("group_by", {})[group_by_key]`. If the `group_by` does not exist it, then you are guaranteed to get a key error because its an empty dictionary. I think the better alternative here would be to do something similar to how the `filter_value` is being handled below: https://github.com/project-koku/koku/blob/30e8dd5bc5467337a177065f47e6cbf4e9e2ea57/koku/api/query_handler.py#L328

Reply:
@blentz completely unrelated to your PR - can we simplify the two for loops here. Right now we are doing: 
```
for group_by_key in parameters.parameters.get('group_by', {}):
            for group_by_value in parameters.parameters['group_by'][group_by_key]:
```
but unless i'm misunderstanding - we could simplify it using: 
```
for group_by_key, group_by_value in parameters.parameters.get('group_by', {}).items():
```
I think this would make it more readable. Does that make sense to you all?  (I know this wasn't introduced by you - but I think I had a hard time understanding what was happening here bc of it)

Reply:
@abaiken - I agree with you, the code itself likely can be simplified.

But, if it's alright with you, I'd like to handle fixing that looping in a separate PR. I'd like to keep this one to just adding the two missing tests.